pkg/prometheus/operator.go

```diff
@@ -1310,6 +1325,94 @@ func (c *Operator) loadBasicAuthSecrets(
 
 }
 
+func (c *Operator) loadBearerTokensFromSecrets(mons map[string]*monitoringv1.ServiceMonitor) (map[string]BearerToken, error) {
+	tokens := map[string]BearerToken{}
+	nsSecretCache := make(map[string]*v1.Secret)
+
+	for _, mon := range mons {
+		for i, ep := range mon.Spec.Endpoints {
+			if ep.BearerTokenSecret.Name == "" {
+				continue
+			}
+
+			sClient := c.kclient.CoreV1().Secrets(mon.Namespace)
+			token, err := getCredFromSecret(
+				sClient,
+				ep.BearerTokenSecret,
+				"bearertoken",
```

Comment:
this might be good to put into a const? Not sure if referenced multiple times.

Reply:
As far as I can tell, `getCredFromSecret` is using this string only ever to log properly. What do you think of instead of passing it in, adding the info later on by wrapping the error? That should make the function call a lot easier on many call sites.